### ISSUE TITLE
[Enhancement] Warn when readonly dependencies are declared but not initialized in ASP.NET Core controllers or services

### ISSUE THREAD

**Issue by Pavan8374** [dotnet/roslyn#78936]
**Brief description:**

Warn when a readonly field or property is declared inside an ASP.NET Core controller or service class but is never initialized, leading to potential NullReferenceExceptions at runtime.

This is a common and frustrating issue, especially for beginners learning dependency injection. The compiler allows it, but the uninitialized object can cause crashes that are hard to debug..

**Languages applicable:**

C# only

```
public class HomeController : Controller {
    private readonly IUserService _userService;

    public HomeController() {
        // Forgot to assign _userService
    }

    public async Task<IActionResult> GetUsers() {
        var users = await _userService.GetAll(); // NullReferenceException at runtime
        return Ok(users);
    }
}.
```

The analyzer should ideally raise a warning on _userService that it is never assigned.

**Additional information:**

- This is not a code style rule per se, but rather a request for a semantic/code quality analyzer.

- It's especially relevant for DI-based applications (ASP.NET Core) where constructor injection is the norm.

- Nullable reference types do not catch this in all cases unless enforced strictly and annotated thoroughly.

- The pattern could be easily recognized by analyzers:

        - A class has a readonly field of reference type
        
        - That field is never assigned in any constructor
        
        - The field is later dereferenced in a method
        
This analyzer could help reduce hard-to-debug runtime exceptions and improve DX for beginners.
